spec/models/time_profile_spec.rb

```diff
@@ -11,6 +11,32 @@
     expect(t.tz).to be_nil
   end
 
+  it "will correctly read the non default initial values" do
+    tp = FactoryBot.create(
+      :time_profile,
+      :description => 'Test1',
+      :days        => [0, 1, 2, 3, 4],
+      :hours       => [0, 1, 2, 5, 6, 8, 9, 10, 11, 12, 19, 20, 21, 22, 23],
+      :tz          => 'Alaksa'
+    )
+    expect(tp.description).to eq('Test1')
+    expect(tp.days).to eq([0, 1, 2, 3, 4])
+    expect(tp.hours).to eq([0, 1, 2, 5, 6, 8, 9, 10, 11, 12, 19, 20, 21, 22, 23])
+    expect(tp.tz).to eq('Alaska')
+  end
+
+  it "will correctly read the non default edited values" do
+    tp = FactoryBot.create(:time_profile)
+    tp.update_attribute(:description, 'Test2')
+    tp.update_attribute(:days, [0, 1, 2, 3, 4])
+    tp.update_attribute(:hours, [0, 1, 2, 5, 6, 8, 9, 10, 11, 12, 19, 20, 21, 22, 23])
+    tp.update_attribute(:tz, 'Hawaii')
+    expect(tp.description).to eq('Test2')
+    expect(tp.days).to eq([0, 1, 2, 3, 4])
+    expect(tp.hours).to eq([0, 1, 2, 5, 6, 8, 9, 10, 11, 12, 19, 20, 21, 22, 23])
+    expect(tp.tz).to eq('Hawaii')
+  end
```

Comment:
While these tests are good and we should keep them for this PR, I think we need an additional tests for when someone calls `profile=` both with a string based Hash and also a symbol based Hash.

Reason is that that is what the API will call, but these tests are only testing the individual getters and setters.